### PR TITLE
tron writes environment variables for the action_runner

### PR DESCRIPTION
**Pull request by drmorr0**
we export four variables, parsed from the run_id

TRON_JOB_NAMESPACE
TRON_JOB_NAME
TRON_RUN_NUM
TRON_ACTION

### CODE REVIEW

File: tron/bin/action_runner.py [hunk 1]
```diff
@@ -103,12 +103,19 @@ def parse_args():
...
 
-def run_command(command):
+def run_command(command, run_id):
+    namespace, job, run_num, action = run_id.split('.')
     return subprocess.Popen(
         command,
         shell=True,
         stdout=subprocess.PIPE,
         stderr=subprocess.PIPE,
+        env=dict(
```
Comment (solarkennedy) [inline tron/bin/action_runner.py]: I'm afraid that this is a breaking change compared to the previous which inherited the original env, including PATH, etc.

Can you make that not the case?

File: tron/bin/action_runner.py [hunk 2]
```diff
@@ -103,12 +103,19 @@ def parse_args():
     return parser.parse_args()
 
 
-def run_command(command):
+def run_command(command, run_id):
+    namespace, job, run_num, action = run_id.split('.')
```
Comment (solarkennedy) [inline tron/bin/action_runner.py]: Can you turn this into a function that returns the env dictionary?

That will be easier to test and change in the future.

Comment (solarkennedy) [reply tron/bin/action_runner.py]: This is too scary also. You are assuming there will only ever be 4 periods. Please make the function more defensive against this.

Also it needs to be in a big try/except. I don't want this split to cause a job not to run.

File: tron/bin/action_runner.py (hunk 1 above)
Comment (drmorr0) [reply tron/bin/action_runner.py]: Done.

Comment (drmorr0) [reply tron/bin/action_runner.py]:
```
drmorr@dev54-uswest1adevc ~tron > tronview --server=http://dev54-uswest1adevc.dev.yelpcorp.com:8089 MASTER.testjob0.1.zeroth
Action Run          : MASTER.testjob0.1.zeroth
State               : succeeded
Node                : drmorr@localhost
Command             : env
Bare command        : env
Start time          : 2019-10-11 15:44:00
End time            : 2019-10-11 15:44:00
Final exit status   : 0
Retry exit statuses : []
Waits for triggers  :
Publishes triggers  : minutely.2019-10-11T15:44
Requirements:

Stdout:
XDG_RUNTIME_DIR=/run/user/4048
LANG=en_US.UTF-8
TRON_ACTION=zeroth
TRON_JOB_NAME=testjob0
SHELL=/bin/zsh
GIT_COMMITTER_EMAIL=drmorr@yelp.com
PWD=/nail/home/drmorr
SSH_CONNECTION=127.0.0.1 39642 127.0.0.1 22
EDITOR=vim
TRON_JOB_NAMESPACE=MASTER

Stderr:

```

File: tron/bin/action_runner.py (hunk 2 above)
Comment (drmorr0) [reply tron/bin/action_runner.py]: Done.

File: tron/bin/action_runner.py [hunk 3]
```diff
@@ -73,6 +73,24 @@ def validate_output_dir(path):
             raise OSError("Could not create output dir %s" % path)
 
 
+def build_environment(run_id):
+    try:
+        namespace, job, run_num, action = run_id.split('.', maxsplit=3)
+    except ValueError:
+        # if we can't parse the run_id, we don't want to abort, so just
+        # set these semi-arbitrarily
+        namespace, job, run_num, action = '', '', '', run_id
```
Comment (solarkennedy) [inline tron/bin/action_runner.py]: `UNKOWN` would be ok I think. That would make it more clear that there was a bug in parsing.

Setting the action to the `run_id` sound bug-prone.

File: tron/bin/action_runner.py [hunk 4]
```diff
@@ -143,7 +162,7 @@ def main():
     args = parse_args()
     validate_output_dir(args.output_dir)
     configure_logging(run_id=args.run_id, output_dir=args.output_dir)
-    proc = run_command(args.command)
+    proc = run_command(args.command, args.run_id)
```
Comment (solarkennedy) [inline tron/bin/action_runner.py]: kwargs please here

**Pull request merged**